pkg/resources/metrics.go

```diff
@@ -246,6 +247,43 @@ func CreateRedisConnectivityAlert(ctx context.Context, client k8sclient.Client,
 	return pr, nil
 }
 
+
+func ReconcilePostgresFreeStorageAlerts(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Postgres) error {
+	if strings.ToLower(inst.Spec.UseClusterStorage) == "true" {
+		logrus.Info("skipping postgres free storage alert creation, useClusterStorage is true")
+		return nil
+	}
+
+	alertName := "PostgresStorageWillFillIn4Hours"
+	ruleName := "postgres-storage-will-fill-in-4-hours"
+	alertDescription := "The postgres instance {{ $labels.instanceID }} for product {{  $labels.productName  }} will run of disk space in the next 4 hours"
+	labels := map[string]string{
+		"severity":    "critical",
+	}
+	alertExp := intstr.FromString("predict_linear(cro_postgres_free_storage_average[1h], 4 * 3600) <= 0")
+
+	// create the rule
+	_, err := reconcilePrometheusRule(ctx, client, ruleName, cr.Namespace, alertName, alertDescription, sopUrlPostgresWillFill, alertFor5Mins, alertExp, labels)
+	if err != nil {
+		return err
+	}
+
+	alertName = "PostgresStorageWillFillIn4Days"
+	ruleName = "postgres-storage-will-fill-in-4-days"
+	alertDescription = "The postgres instance {{ $labels.instanceID }} for product {{  $labels.productName  }} will run of disk space in the next 4 days"
+	labels = map[string]string{
+		"severity":    "critical",
+	}
+	alertExp = intstr.FromString("predict_linear(cro_postgres_free_storage_average[6h], 4 * 24 * 3600) <= 0")
```

Comment:
any way to get this `cro_postgres_free_storage_average` from cro as a const https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/controller/cloudmetrics/cloudmetrics_controller.go#L28

Reply:
never mind its not public

Reply:
Yes and No, we will run into the same incompatible issues due to the SDK bump. So it just would not be worth slowing down development of this for a string. That work is tracked here -> https://issues.redhat.com/browse/INTLY-9495

Reply:
:heavy_check_mark: :godmode:

Reply:
Looks like the priority of INTLY-9495 is getting bumped